subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_spi_stm32.c

```diff
@@ -0,0 +1,715 @@
+			.source_burst_length = 1, /* SINGLE transfer */                            \
+			.dest_burst_length = 1,	  /* SINGLE transfer */                            \
+			.channel_priority = STM32_DMA_CONFIG_PRIORITY(                             \
+				DT_DMAS_CELL_BY_NAME(id, dir, channel_config)),                    \
+			.dma_callback = dma_callback,                                              \
+			.block_count = 2,                                                          \
+	},                                                                                         \
+	.fifo_threshold =                                                                          \
+		STM32_DMA_FEATURES_FIFO_THRESHOLD(DT_DMAS_CELL_BY_NAME(id, dir, features)),
+
+#define STM32_SPI_INIT(id)                                                                         \
+	PINCTRL_DT_DEFINE(id);                                                                     \
+	static const struct stm32_pclken pclken[] = STM32_DT_CLOCKS(id);                           \
+                                                                                                   \
+	static struct ec_host_cmd_spi_cfg ec_host_cmd_spi_cfg = {                                  \
+		.spi = (SPI_TypeDef *)DT_REG_ADDR(id),                                             \
+		.pclken = pclken,                                                                  \
+		.pclk_len = DT_NUM_CLOCKS(id),                                                     \
+		.pcfg = PINCTRL_DT_DEV_CONFIG_GET(id),                                             \
+	};                                                                                         \
+                                                                                                   \
+	static struct dma_stream dma_rx = {SPI_DMA_CHANNEL_INIT(id, rx, RX, PERIPHERAL, MEMORY)};  \
+	static struct dma_stream dma_tx = {SPI_DMA_CHANNEL_INIT(id, tx, TX, MEMORY, PERIPHERAL)}
+
+STM32_SPI_INIT(DT_CHOSEN(zephyr_host_cmd_spi_backend));
+
+#define EC_HOST_CMD_SPI_DEFINE(_name)                                                              \
+	static struct ec_host_cmd_spi_ctx _name##_hc_spi = {                                       \
+		.dma_rx = &dma_rx,                                                                 \
+		.dma_tx = &dma_tx,                                                                 \
+		.spi_config = &ec_host_cmd_spi_cfg,                                                \
+	};                                                                                         \
+	struct ec_host_cmd_backend _name = {                                                       \
+		.api = &ec_host_cmd_api,                                                           \
+		.ctx = (struct ec_host_cmd_spi_ctx *)&_name##_hc_spi,                              \
+	}
+
+static inline uint32_t dma_source_addr(SPI_TypeDef *spi)
+{
+#ifdef CONFIG_SOC_SERIES_STM32H7X
```

Comment:
Please replace all
```
#ifdef CONFIG_SOC_SERIES_STM32H7X
```
by:
```
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
```
It will enable support on U5, MP1 and H5 series (and upcoming series with similar SPI HW block)

Reply:
I'm using `DT_NODE_HAS_COMPAT` instead of `DT_HAS_COMPAT_STATUS_OKAY` because the spi chosen node relies on the status `disabled` not to be handled by the default stm32 spi driver.